### DISCUSSION TITLE
Do not replace the os.sep

### DISCUSSION THREAD

**Discussion by Harding-Stardust** [arizvisa/ida-minsc#188] [Q&A] [answered]
```return res.replace(os.sep, '/')```
should be changed to ```return res```

Why is there a change to / as separator? This will be strange on Windows.

https://github.com/arizvisa/ida-minsc/blob/af07f167a0992aed3f3a2a1670593880baa57820/base/database.py#L141

**Comment by arizvisa** [comment]
This is done because python treats forward slashes as an alternative separator. Thus if you're using python's file APIs, they have the same results. Plus, since python's strings are always backslash-escaped (unless they're `r''`-prefixed) you can avoid having to type twice-as-many path separators. Essentially, this normalizes the path so that your code works the same regardless of the platform that it is running on.

**Comment by arizvisa** [comment]
Btw, only marked this as a "wontfix" due to it not appearing to be a bug and rather just a question that's likely better placed under "Discussions" (despite nobody using that shit). To be clear, I don't mind answering questions here though. So don't let that deter you.

**Comment by Harding-Stardust** [comment]
> This is done because python treats forward slashes as an alternative separator. Thus if you're using python's file APIs, they have the same results. Plus, since python's strings are always backslash-escaped (unless they're `r''`-prefixed) you can avoid having to type twice-as-many path separators. Essentially, this normalizes the path so that your code works the same regardless of the platform that it is running on.

I understand that using a literal \ in strings in Python has to be escaped (or the string can be r prefixed) but in this context it doesn't matter as the string containing the \ character wont cause any problems.

**Comment by arizvisa** [comment, marked as answer]
Still, its purpose is to normalize the database path. This is useful when you're emitting a path cross-platform and is purely so that you concatenate or split the path without having to depend on a module. This is useful when you attach a python script to a database that serializes all your changes to a directory that is relative to the location of your database. This enables that same script to work regardless of the platform it is running on and allows you to avoid depending on anything within the `os` module.

Since it doesn't cause any issues with the Python interpreter on the windows platform, I'm still unsure about its strangeness and how it directly affects your scripts.
